src/app/js/formats/vega/component/tree-map/TreeMapData.js

```diff
@@ -0,0 +1,166 @@
+export default class TreeMapData {
+    /**
+     * Raw data to be filtered and transformed
+     * @type {Array<{source:string, target: string, weight: number}>}
+     */
+    data;
+    /**
+     * Nodes and leaves ids counter
+     * @type {number}
+     */
+    idIncrement;
+    /**
+     * List of nodes and leaves ids
+     * @type {Map<string, number>}
+     */
+    ids;
+    /**
+     * Raw list of nodes and leaves ready to be filtered and transformed
+     * @type {Map<string, {parent: number?, size: number?}>}
+     */
+    rawNodesAndLeaves;
+    /**
+     * Formatted list of nodes and leaves ready to be cleanup and returned
+     * @type {Map<number, {id: number, name: string, parent: number?, size: number?}>}
+     */
+    formattedNodesAndLeaves;
+    /**
+     * Filtered nodes use to create the final transformed data
+     * @type {Set<number>}
+     */
+    filteredNodes;
```

Comment:
<!-- Suggestions propose improvements to the current subject. It's important to be explicit and clear on what is being suggested and why it is an improvement. Consider using patches and the blocking or non-blocking decorations to further communicate your intent.  -->
**suggestion:** Precise in comments that filtered nodes are not leaves‏